src/legacy/core_plugins/data/public/filter/filter_manager/filter_state_manager.test.ts

```diff
@@ -164,4 +155,25 @@ describe('filter_state_manager', () => {
       sinon.assert.calledOnce(globalStateStub.save);
     });
   });
+
+  describe('bug fixes', () => {
+    /*
+     ** This test is here to reproduce a bug where a filter manager update
+     ** would cause filter state manager detects those changes
+     ** And triggers *another* filter manager update.
+     */
+    test('should NOT re-trigger filter manager', async done => {
+      const f1 = getFilter(FilterStateStore.APP_STATE, false, false, 'age', 34);
+      filterManager.setFilters([f1]);
+      const setFiltersSpy = sinon.spy(filterManager, 'setFilters');
+
+      f1.meta.negate = true;
+      await filterManager.setFilters([f1]);
+
+      setTimeout(() => {
```

Comment:
Why does this need to be wrapped in a `setTimeout` if you're already `await`ing above?

Reply:
The reason for this is that `filter_state_manager` picks up the changes `setFilter` makes, in a `setInterval` loop.

Obviously this is a temporary solution until state management \ url filter sync is not done in Angular.